### ISSUE TITLE
aws_codebuild_project is missing NO_SOURCE source type

### ISSUE THREAD

**Issue by atupeli** [terraform-providers/terraform-provider-aws#6076]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Add "NO_SOURCE" option to allowed types in aws_codebuild_project source block. "NO_SOURCE" is supported by CodeBuild API [ProjectSource](https://docs.aws.amazon.com/codebuild/latest/APIReference/API_ProjectSource.html). Notice that "buildspec" property is required when "NO_SOURCE" type is used.

### New or Affected Resource(s)

* aws_codebuild_project

### Potential Terraform Configuration

```hcl
resource "aws_codebuild_project" "test" {
  ...

  source {
    type = "NO_SOURCE"
    buildspec = "..."
  }

  ...

}
```

### References

* https://www.terraform.io/docs/providers/aws/r/codebuild_project.html
* https://docs.aws.amazon.com/codebuild/latest/APIReference/API_ProjectSource.html

**Comment by tkbky** [comment]
I've made a [PR](https://github.com/terraform-providers/terraform-provider-aws/pull/6140) for this.

**Comment by bflad** [comment]
Support for this has been merged and will release with version 1.41.0 of the AWS provider, likely middle of this week. 👍

**Comment by bflad** [comment]
This has been released in [version 1.41.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1410-october-18-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Comment by github4es** [comment]
Has anybody come across a sample where the source is set to NO_SOURCE and one has to provide the "Buildspec"?  I'm loooking for an example. 

Given that the buildspec is set as a YAML definition that could span 100+ lines, how does one embed that in the terraform definition?  

`resource "aws_codebuild_project" "test" {
  ...

  source {
    type = "NO_SOURCE"
    buildspec = "version: 0.2
    env:
     variables:
      APP_NAME: "sample-project"
      parameter-store:
          NPM_REGISTRY: "/abc/npm"
      phases:
        install:
         runtime-versions:
            nodejs: 10
  }
}`
